### ISSUE TITLE
PrestoExchangeSource HttpClient histogram metric is blocking query

### ISSUE THREAD

**Issue by karteekmurthys** [prestodb/presto#23338]
[PrometheusStatsReporter](https://github.com/prestodb/presto/pull/22360) updates worker stats synchronously. This is fine as most of the stats are updated through PeriodicTaskManager in a separate thread periodically without blocking queries.

But [this specific metric](https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp#L117-L119) is updated in the query path and is updated way too frequently.
```
        RECORD_HISTOGRAM_METRIC_VALUE(
            kCounterHttpClientPrestoExchangeOnBodyBytes, bufferBytes);
      });
```
The Q67 of SF10K TPCDS workload takes several hours to finish since we merged PrometheusStatsReporter. We are working on how to improve this in the stats reporter but we should also avoid reporting metrics in the query path if possible.

Request to review if `presto_cpp.http.client.presto_exchange_source.on_body_bytes` can be removed or made optional for tracking.

We tested Q67 at SF10K by removing this metric and the query completes successfully.

We also have a workaround of updating [histogram asynchronously](https://github.com/prestodb/presto/commit/437a333fd9b024e9c66efd9c84de94a35663e3e9) in PrometheusStatsReporter. This solution was tested successfully as well against Q67

**Comment by karteekmurthys** [comment]
cc: @majetideepak @amitkdutta

**Comment by karteekmurthys** [comment]
@pramodsatya gathered some stats total time spent different locations in Presto CPP where RECORD_HISTOGRAM is called. The Q67 query was run for over an hour and `PrestoExchangeSource.cpp:117`  took total of 3256278906 ms in hour of query run.


```
reg/add   function                               line num
Add       addRootPool                            Memory.cpp:248                          76
          clearThread                            Driver.h:153                        166326
          operator()                             ExchangeClient.cpp:163               87166
                                                 ExchangeClient.cpp:167               61760
                                                 ExchangeClient.cpp:170               31759
                                                 FileHandle.cpp:64                       45
                                                 PrestoExchangeSource.cpp:117    3256278906
          runInternal                            Driver.cpp:522                      348599
          updatePrestoExchangeSourceMemoryStats  PeriodicTaskManager.cpp:268          18196
          ~MemoryPoolImpl                        MemoryPool.cpp:462                      45
          ~ScopedArbitration                     SharedArbitrator.cpp:880              1110
                                                 SharedArbitrator.cpp:910                12
```

**Comment by jaystarshot** [comment]
What if the implementation instead created a subprocess like the exposer mentioned https://github.com/jupp0r/prometheus-cpp. Resources can be controlled when starting the exposer

**Comment by jaystarshot** [comment]
I don't think this will help directly here^ since I could repro this internally.

**Comment by majetideepak** [comment]
Resolved with https://github.com/prestodb/presto/pull/23357
